controllers/secretproviderclasspodstatus_controller.go

```diff
@@ -0,0 +1,278 @@
+			errs = append(errs, fmt.Errorf("failed to check if secret %s exists, err: %+v", secretObj.SecretName, err))
+			continue
+		}
+		funcs := []func() (bool, error){}
+
+		if !exists {
+			secretType := getSecretType(secretObj.Type)
+			datamap := make(map[string][]byte)
+
+			for _, data := range secretObj.Data {
+				if len(data.ObjectName) == 0 {
+					logger.Errorf("object name in data is empty at index %d for secret %s", idx, secretObj.SecretName)
+					errs = append(errs, fmt.Errorf("object name in data is empty at index %d for secret %s", idx, secretObj.SecretName))
+					continue
+				}
+				if len(data.Key) == 0 {
+					logger.Errorf("key in data is empty at index %d for secret %s", idx, secretObj.SecretName)
+					errs = append(errs, fmt.Errorf("key in data is empty at index %d for secret %s", idx, secretObj.SecretName))
+					continue
+				}
+				file, ok := files[data.ObjectName]
+				if !ok {
+					logger.Errorf("file matching objectName %s not found for secret %s", data.ObjectName, secretObj.SecretName)
+					continue
+				}
+				logger.Infof("file matching objectName %s found for key %s, secret %s", data.ObjectName, data.Key, secretObj.SecretName)
+				content, err := ioutil.ReadFile(file)
+				if err != nil {
+					logger.Errorf("failed to read file %s, err: %v", data.ObjectName, err)
+					return ctrl.Result{}, status.Error(codes.Internal, err.Error())
+				}
+				datamap[data.Key] = content
+				if secretType == corev1.SecretTypeTLS {
+					c, err := getCertPart(content, data.Key)
+					if err != nil {
+						logger.Errorf("failed to get cert data from file %s, err: %v for secret: %s", file, err, secretObj.SecretName)
+						return ctrl.Result{RequeueAfter: 5 * time.Second}, status.Error(codes.Internal, err.Error())
+					}
+					datamap[data.Key] = c
+				}
+			}
+
+			createFn := func() (bool, error) {
+				if err := r.createK8sSecret(ctx, secretObj.SecretName, req.Namespace, datamap, secretType); err != nil {
+					logger.Errorf("failed createK8sSecret, err: %v for secret: %s", err, secretObj.SecretName)
+					return false, nil
+				}
+				return true, nil
+			}
+			funcs = append(funcs, createFn)
+		}
+
+		// patch the secret with the owner reference
+		patchFn := func() (bool, error) {
+			if err := r.patchSecretWithOwnerRef(ctx, secretObj.SecretName, req.Namespace, &spcPodStatus); err != nil {
+				logger.Errorf("failed to set owner ref for secret, err: %+v", err)
+				return false, nil
+			}
+			return true, nil
+		}
+
+		funcs = append(funcs, patchFn)
+		for _, f := range funcs {
+			if err := wait.ExponentialBackoff(wait.Backoff{
+				Steps:    5,
+				Duration: 1 * time.Millisecond,
+				Factor:   1.0,
+				Jitter:   0.1,
+			}, f); err != nil {
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, err
+			}
+		}
+	}
+
+	if len(errs) > 0 {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	logger.Info("reconcile complete")
+	return ctrl.Result{RequeueAfter: 5 * time.Minute}, nil
```

Comment:
why is this queue again after 5mins?

Reply:
This is done to periodically reconcile and ensure secrets that were created still exists and the owner reference for the spc pod status exists as expected

Reply:
oh duh. this is the loop :)